stdlib/public/runtime/BytecodeLayouts.cpp

```diff
@@ -281,9 +281,12 @@ static void weakDestroy(const Metadata *metadata, LayoutStringReader1 &reader,
 static void unknownDestroy(const Metadata *metadata,
                            LayoutStringReader1 &reader, uintptr_t &addrOffset,
                            uint8_t *addr) {
-  void *object = *(void**)(addr + addrOffset);
+  uintptr_t object = *(uintptr_t *)(addr + addrOffset);
   addrOffset += sizeof(void*);
-  swift_unknownObjectRelease(object);
+  if (_swift_abi_ObjCReservedBitsMask == 0) {
```

Comment:
It would be good to comment these so we can remember why this masking is predicated on 0 reserved bits. Maybe extract it into a little helper function with a more descriptive name.